### PR TITLE
Add parameter(learning_rate) in NoamDecay

### PR DESCRIPTION
**Pull request by Aurelius84**
+ Add parameter(learning_rate) in NoamDecay, default 1.0

`NoamDecay` is a strategy for learning rate attenuation which is used in Transformer(See《Attension is all you need》). The calculation of decayed learning rate as follows, and it doesn't dependent on origin `learning_rate` and is only relates with `current_step` and `warmup_steps`.

image
image

But, in some cases, `base_lr` may be applied to multiple the returned valued of `NoamDecay`, just like [`CosineDecay`](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/dygraph_cn/CosineDecay_cn.html#cosinedecay), [ExponentialDecay](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/dygraph_cn/ExponentialDecay_cn.html#exponentialdecay) and so on.

Unlike `CosineDecay`, the `learning_rate` is not a required parameter because `NoamDecay` doesn't dependent on it. So we set it 1.0 by default which is compatible with the previous usage.

image
image

Zh-CN doc: https://github.com/PaddlePaddle/FluidDoc/pull/1940

### CODE REVIEW

File: python/paddle/fluid/dygraph/learning_rate_scheduler.py [hunk 1]
```diff
@@ -550,16 +553,24 @@ class NoamDecay(LearningRateDecay):
                   parameter_list = emb.parameters())
     """
 
-    def __init__(self, d_model, warmup_steps, begin=1, step=1, dtype='float32'):
+    def __init__(self,
+                 d_model,
+                 warmup_steps,
+                 learning_rate=1.0,
```
Comment (lanxianghit) [inline python/paddle/fluid/dygraph/learning_rate_scheduler.py]: 新增参数为什么加在参数列表中间？会引起不兼容风险吧

File: python/paddle/fluid/dygraph/learning_rate_scheduler.py [hunk 2]
```diff
@@ -526,6 +526,9 @@ class NoamDecay(LearningRateDecay):
             it's a tensor with shape [1] and the data type can be int32 or int64. The type can also be python int.
         warmup_steps(Variable|int): The number of warmup steps. A super parameter. If type is Variable, 
             it's a tensor with shape [1] and the data type can be int32 or int64. The type can also be python int.
+        learning_rate(Variable|float): The initial learning rate. If the type
+            is Variable, it's a tensor with shape [1], the data type can be
+            float32 or float64. It also can be set to python int number. Default 1.0
```
Comment (lanxianghit) [inline python/paddle/fluid/dygraph/learning_rate_scheduler.py]: > It also can be set to python int number
支持int类型？是否要写在前面的括号里？

File: python/paddle/fluid/tests/unittests/test_learning_rate_scheduler.py [hunk 3]
```diff
@@ -89,6 +89,34 @@ def cosine_decay(global_step, learning_rate, step_each_epoch, epochs):
...
+                                          learning_rate)
+                fluid_result = lr()
+
+                self.assertAlmostEqual(
+                    right_result,
+                    fluid_result[0],
+                    msg='Failed lr scheduler in step {0}, Python result is {1}, Fluid result is {2}'.
+                    format(step, right_result, fluid_result[0]))
+
+
```
Comment (lanxianghit) [inline python/paddle/fluid/tests/unittests/test_learning_rate_scheduler.py:119]: 除单测外，最好在实际模型上跑一个验证结果

File: python/paddle/fluid/dygraph/learning_rate_scheduler.py (hunk 1 above)
Comment (Aurelius84) [reply python/paddle/fluid/dygraph/learning_rate_scheduler.py]: 后面三个`begin`/`step`/`dtype`是基类`LearnintRateDecay`的默认参数。放在中间，之前的考虑是：与其他Decay接口的后续参数保持一致。若用户高阶使用`begin`/`step`且未指定参数名时，确实会有兼容风险，我调整下顺序。

File: python/paddle/fluid/dygraph/learning_rate_scheduler.py (hunk 2 above)
Comment (Aurelius84) [reply python/paddle/fluid/dygraph/learning_rate_scheduler.py]: 好的，我更新下括号里的参数类型

File: python/paddle/fluid/tests/unittests/test_learning_rate_scheduler.py (hunk 3 above)
Comment (Aurelius84) [reply python/paddle/fluid/tests/unittests/test_learning_rate_scheduler.py:119]: 在Transoformer动态图模型上自测小数据集10个batch迭代，符合预期。

**Review by lanxianghit — APPROVED** [review]
LGTM

**Review by Boyan-Liu — APPROVED** [review]
